cassiopeia/solver/CassiopeiaSolver.py

```diff
@@ -15,38 +15,16 @@ class CassiopeiaSolver(abc.ABC):
     """
     CassiopeiaSolver is an abstract class that all inference algorithms derive
     from. At minimum, all CassiopeiaSolver subclasses will store a character
-    matrix and implement a solver procedure.
+    matrix and implement a solver procedure.        
 
-    Args:
-        character_matrix: A character matrix of observed character states for
-            all samples
-        missing_char: The character representing missing values
-        meta_data: Any meta data associated with the samples
-        priors: Prior probabilities of observing a transition from 0 to any
-            character state
-
-    Attributes:
-        character_matrix: The character matrix describing the samples
-        missing_char: The character representing missing values
-        meta_data: Data table storing meta data for each sample
-        priors: Prior probabilities of character state transitions
-        tree: The tree built by `self.solve()`. None if `solve` has not been
-            called yet
     """
 
     def __init__(
         self,
-        character_matrix: pd.DataFrame,
-        missing_char: int,
-        meta_data: Optional[pd.DataFrame] = None,
-        priors: Optional[Dict] = None,
+        prior_transformation: str = "negative_log"
     ):
 
-        self.character_matrix = character_matrix
-        self.missing_char = missing_char
-        self.meta_data = meta_data
-        self.tree = None
-        self.priors = priors
+        self.prior_transformation = prior_transformation
 
     @abc.abstractmethod
     def solve(self):
```

Comment:
Should be `solve(self, cassiopeia_tree: CassiopeiaTree) -> None`

Reply:
I was under the impression if a method does not return anything we don't need to have `-> None`. Is this not technically correct?

Reply:
Ah, the highlight was the missing `cassiopeia_tree: CassiopeiaTree` arguent, not the `None` return value 😂 . You are probably right about `None`, I'm just paranoid and add it too, no need to fix that.